### DISCUSSION TITLE
Does Dynamoose Support Write Sharding for DynamoDB?

### DISCUSSION THREAD

**Discussion by cckn** [dynamoose/dynamoose#1643] [Q&A]
Hi Dynamoose Team,

I'm looking for ways to handle scenarios in DynamoDB where a single partition key has to accommodate a large number of items. AWS recommends write sharding as a solution (as seen in their [partition key sharding guide](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-partition-key-sharding.html) and [Amazon DynamoDB Labs sharding article](https://amazon-dynamodb-labs.com/design-patterns/ex3gsisharding.html)).

Does Dynamoose have any features or best practices for implementing write sharding in this context? Any advice or shared experiences would be greatly appreciated.

Thanks!

P.S. I am currently exploring various libraries to better understand which one aligns best with my project's needs. In pursuit of gathering diverse insights, I've posed the same question in the Typedorm repository.

**Comment by fishcharlie** [comment]
@cckn You should be able to use the [`default`](https://dynamoosejs.com/guide/Schema#default-value--function--async-function) setting for this in your schema. 

Something like this maybe?

```js
const schema = {
    "id": {
        "type": String,
        "default": () => `${new Date().getFullYear()}-${new Date().getMonth()}-${new Date().getDay()}.${randomNumber()}`,
    },
    "name": String
};

const Entry = new Model(schema);

const entry = new Entry({
    // id will be automatically generated based on the `default` function above
    name: "John Doe"
});

// Random number between 1-200
function randomNumber() {
    return Math.floor(Math.random() * 200) + 1;
}
```

Note: I haven't actually tested this code. Just wrote it off the top of my head. But just meant to illustrate an example and not meant to be used in production.

Hopefully this helps!
